packages/blocks/src/root-block/edgeless/services/tools-manager.ts

```diff
@@ -290,33 +313,53 @@ export class EdgelessToolsManager {
   };
 
   private _onContainerPointerDown = (e: PointerEventState) => {
-    if (!isMiddleButtonPressed(e.raw)) {
-      if (this.doc.readonly) return;
+    const pointEvt = e.raw;
+    const metaKeyPressed = IS_MAC ? pointEvt.metaKey : pointEvt.ctrlKey;
 
-      return this.currentController.onContainerPointerDown(e);
+    if (
+      isMiddleButtonPressed(pointEvt) ||
+      isRightButtonPressed(pointEvt) ||
+      metaKeyPressed
+    ) {
+      const isRightButton = isRightButtonPressed(pointEvt);
+      const targetTool = (
+        isRightButton || metaKeyPressed
```

Comment:
@doouding 
Here may should be "isRightButton && metaKeyPressed" if you meant rightbutton-dragging with ctrl or command (mac) pressed.

Reply:
Actuallly not, here I mean right button dragging OR meta key pressed.

Reply:
> Actuallly not, here I mean right button dragging OR meta key pressed.

Oh I thought right button dragging could pan too. I take back.